From 4904bedd493a095ed18f21673349cccd8b0022b9 Mon Sep 17 00:00:00 2001
From: frq09432 <nicolas.louboutin@st.com>
Date: Thu, 27 Jun 2019 13:52:16 +0200
Subject: [PATCH] stm32mp1: add fastboot boot instance variable

  Add new fastboot dedicated variable to allow getting back boot instance (used to manage provisioning)
---
 arch/arm/mach-stm32mp/cpu.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-stm32mp/cpu.c b/arch/arm/mach-stm32mp/cpu.c
index 8d8ade3..1fc3f2b 100644
--- a/arch/arm/mach-stm32mp/cpu.c
+++ b/arch/arm/mach-stm32mp/cpu.c
@@ -405,6 +405,9 @@ static void setup_boot_mode(void)
 		sprintf(cmd, "%d", instance);
 		env_set("boot_device", "mmc");
 		env_set("boot_instance", cmd);
+#ifdef CONFIG_FASTBOOT_FLASH_MMC
+		env_set("fastboot.boot_instance", cmd);
+#endif
 		break;
 	case BOOT_FLASH_NAND:
 		env_set("boot_device", "nand");
-- 
2.7.4

